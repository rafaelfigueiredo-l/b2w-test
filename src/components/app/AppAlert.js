import React from "react";
import styled from "styled-components";


const AppAlert = (props) => {

    console.log(props);

    if(false === props.alertError.status || null === props.alertError.status){
        return null;
    }

    return (
        <>
        <Alert className={`alert alert-${props.alertError.type}`}><span>{props.alertError.message}</span></Alert>
        </>
    );
};

const Alert = styled.div`
  background:#ccc;
  padding:3px 20px;
  border-radius:20px;

  &.alert-error{
      background:red;
  }
  &.alert-message{
    background:#616be2;
  }
  &.alert-success{
    background:#63e261;
  }

`;

export default AppAlert;
