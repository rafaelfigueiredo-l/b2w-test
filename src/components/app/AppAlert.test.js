import AppAlert from './AppAlert';

describe('<AppAlert />', () => {
    it('renders without crashing', () => {
        let obj = {
            status: true,
            type: 'Some Type',
            message: 'Some Message',
            component: 'Some Component',
        };

        const div = document.createElement('div');
        ReactDOM.render(<AppAlert alertError={obj} />, div);
        ReactDOM.unmountComponentAtNode(div);
      });

    it('Snapshot testing', () => {
        let obj = {
            status: true,
            type: 'Some Type',
            message: 'Some Message',
            component: 'Some Component',
        };
        const wrapper = shallow(<AppAlert alertError={obj} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('Snapshot false || null ? should return null', () => {
            let obj = {
                status: false,
                type: 'Some Type',
                message: 'Some Message',
                component: 'Some Component',
            };
            const wrapper = shallow(<AppAlert alertError={obj} />);
            expect(wrapper).toMatchSnapshot();
    });
});