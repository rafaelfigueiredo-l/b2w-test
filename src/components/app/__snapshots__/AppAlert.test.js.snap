// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppAlert /> Snapshot false || null ? should return null 1`] = `""`;

exports[`<AppAlert /> Snapshot testing 1`] = `
<styled.div
  className="alert alert-Some Type"
>
  <span>
    Some Message
  </span>
</styled.div>
`;
